keps/sig-cli/5295-kyaml/README.md

```diff
@@ -0,0 +1,1428 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [X] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes to add a new output format for kubectl,
+called "KYAML"
+(as in `kubectl get -o kayml ...`).
+This format is a strict subset (aka "dialect") of standard YAML,
+and so should be parseable by the existing ecosystem.
+This dialect seeks to emphasize syntactical choices which avoid
+many of the most common traps in YAML.
+For example, unlike standard YAML output,
+this dialect is not whitespace-sensitive,
+which makes it vastly easier to patch correctly in things like Helm charts.
```

Comment:
I brought up this KEP in the Helm Weekly Developer Meeting which I regularly participate.
I'd like to get Helm release manage @mattfarina 's take on all of it noted here.

My take is that while you are saying this is optional, you really are suggesting that people use/recommend/document an entirely new format, that while technically valid YAML, will not be recognized by most users as such.
YAML has been in k8s since the start. That's 10+ years of establishment that is trying to be changed.
And for what I see is: trading a couple common issues for a whole new set of them. 

cc: @banjoh @scottrigby @gjenkins8 from https://github.com/helm/helm

Reply:
I'm fine to take feedback, but I don't think we should block merge on that.  The KEP window closes soon, and I am travelling in a week, so I don't want to miss it on technicality.

Reply:
When does the KEP window close?